File: hw/sanchez/u3/hw24AbstractFactory/AbstractFactoryJs/winFactory.js
const GUIFactory = require("./guiFactory");
const WinButton = require("./winButton");
const WinMenu = require("./winMenu");

class WinFactory extends GUIFactory {
    createButton() {
        return new WinButton();
    }

    createMenu() {
        return new WinMenu();
    }
}

module.exports = WinFactory;

File: hw/palomo/u3/hw23AbstracFactory/AbstracFactory.py/ec_edu_espe/model/win_factory.py
from ec_edu_espe.model.gui_factory import GUIFactory
from ec_edu_espe.model.win_button import WinButton
from ec_edu_espe.model.win_menu import WinMenu


class WinFactory(GUIFactory):
    """ConcreteFactory: crea la familia de productos con estilo Windows."""

    def create_button(self) -> WinButton:
        return WinButton()

    def create_menu(self) -> WinMenu:
        return WinMenu()